src/axmysql/sagasnapshot.go

```diff
@@ -0,0 +1,86 @@
+package axmysql
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/jmalloc/ax/src/ax/persistence"
+	"github.com/jmalloc/ax/src/ax/saga"
+)
+
+// SnapshotRepository is an interface for loading and saving snapshots of
+// eventsourced saga data.
+type SnapshotRepository struct{}
+
+// LoadSagaSnapshot loads the latest available snapshot from the store.
+func (SnapshotRepository) LoadSagaSnapshot(
+	ctx context.Context,
+	ptx persistence.Tx,
+	id saga.InstanceID,
+) (saga.Instance, bool, error) {
+	var (
+		i           saga.Instance
+		contentType string
+		data        []byte
+	)
+
+	err := sqlTx(ptx).QueryRowContext(
+		ctx,
+		`SELECT
+			instance_id,
+			revision,
+			content_type,
+			data
+		FROM saga_snapshot
+		WHERE instance_id = ?
+		ORDER BY revision DESC
+		LIMIT 1`,
+		id,
+	).Scan(
+		&i.InstanceID,
+		&i.Revision,
+		&contentType,
+		&data,
+	)
+
+	if err == sql.ErrNoRows {
+		return saga.Instance{}, false, nil
+	}
+
+	if err != nil {
+		return saga.Instance{}, false, nil
+	}
+
+	i.Data, err = saga.UnmarshalData(contentType, data)
+
+	return i, true, err
+}
+
+// SaveSagaSnapshot saves a snapshot to the store.
+func (SnapshotRepository) SaveSagaSnapshot(
+	ctx context.Context,
+	tx persistence.Tx,
```

Comment:
You called this `ptx` in the `LoadSagaSnapshot()` method. Rename one for consistency?

Reply:
Yeah, could go either way. I did this because in the *implementation* `ptx` is the one that you use the least.  They don't *have* to match really.